packages/graphql/src/schema/subscriptions/generate-subscription-types.ts

```diff
@@ -31,47 +34,77 @@ export function generateSubscriptionTypes({
 }) {
     const subscriptionComposer = schemaComposer.Subscription;
 
-    nodes.forEach((node) => {
-        const composeNode = schemaComposer.getOTC(node.name);
+    const eventTypeEnum = schemaComposer.createEnumTC(EventType);
 
+    nodes.forEach((node) => {
+        const eventPayload = generateEventPayloadType(node, schemaComposer);
+        const where = generateSubscriptionWhereType(node, schemaComposer);
         const subscribeOperation = node.rootTypeFieldNames.subscribe;
 
         const nodeCreatedEvent = schemaComposer.createObjectTC({
             name: `${node.name}CreatedEvent`,
             fields: {
-                [subscribeOperation.created]: composeNode,
+                event: {
+                    type: eventTypeEnum.NonNull,
+                    resolve: () => EventType.getValue("CREATE"),
+                },
+                [`created${node.name}`]: {
+                    type: eventPayload.NonNull,
+                    resolve: (source: SubscriptionsEvent) => source.properties.new,
+                },
             },
         });
 
         const nodeUpdatedEvent = schemaComposer.createObjectTC({
             name: `${node.name}UpdatedEvent`,
             fields: {
-                [subscribeOperation.updated]: composeNode,
+                event: {
+                    type: eventTypeEnum.NonNull,
+                    resolve: () => EventType.getValue("UPDATE"),
+                },
+                previousState: {
+                    type: eventPayload.NonNull,
```

Comment:
question: Will this work fine with nodes only having relationships?

Reply:
Following our conversation this morning, I believe the right approach is to not add the payload types if they have no actual value. This is now implemented.